cabal-install/src/Distribution/Client/Dependency.hs

```diff
@@ -848,7 +829,7 @@ resolveDependencies platform comp pkgConfigDB solver params =
                     noReinstalls
                     shadowing
                     strFlags
-                    allowBootLibs
+                    _allowBootLibs
```

Comment:
Can this field be removed from `DepResolverParams`? We should avoid the situation where we have some user visible toggle that end up having no effect.

Reply:
ah, alright, there's
```
        if asBool (depResolverAllowBootLibInstalls params)
          then params
          else dontUpgradeNonUpgradeablePackages params
```
right under it. I find the dataflow in this function a bit confusing (To be clear, it's always been :relaxed:)